### ISSUE TITLE
Need gce_facts module to get Google instance information.

### ISSUE THREAD

**Issue by Rahul-CSI** [ansible/ansible#35181]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
!component

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/centos/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### CONFIGURATION
[centos@localhost csi-ansible]$ ansible-config dump --only-changed
[centos@localhost csi-ansible]$

##### SUMMARY
Currently we do not have a way to retrieve the info for a node.
If I wish to do some processing based on existing node info, like disks attached.
I would like to have a gce_facts module which would do that.

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/network/cloudengine/ce_facts.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/cloudengine/ce_facts.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @CloudEngine-Ansible @QijunPan
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
@Rahul-CSI Thanks for reporting this issue. Could you please try [gcp_compute_instance_facts](https://github.com/ansible/ansible/lib/ansible/modules/cloud/google/gcp_compute_instance_facts.py) and let us know if it solves your requirement ? Thanks

needs_info

**Comment by Akasurde** [comment]
!component =lib/ansible/modules/cloud/google/gcp_compute_instance_facts.py

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/google/gcp_compute_instance_facts.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/google/gcp_compute_instance_facts.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @erjohnso @googlecloudplatform @rambleraptor
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Rahul-CSI This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Rahul-CSI You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by wojtas911** [comment]
Hi,
I didn't like the idea to open a new issue since my problem is similar. I have learned a lesson from AWS and now since we switched to google compute I'm adding tag/labels as much as I can.
So my playbooks in AWS was depending on those labels and i like to continue that on GCP.
I'm looking for quite a log time some module that will be able to do:
curl -sH "Metadata-Flavor: Google" 'http://metadata.google.internal/computeMetadata/v1/?recursive=True'|jq .instance.attributes.role
"haproxy"
or even easier:
curl -H "Metadata-Flavor: Google" 'http://metadata/computeMetadata/v1/instance/attributes/role'
haproxy
BUT my digging through the vastness of the internet didn't get any result.
Is there any module that i can use the same as aws ec2_tag
- debug: var={{ ec2_tag_role }}
